src/CADL.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -326,19 +427,32 @@ function loadOperation(
             models,
             enums
         );
+        let defaultValue = undefined;
+        const value = getDefaultValue(cadlType);
+        if (value) {
+            defaultValue = {
+                Type: inputType,
+                Value: value
+            } as InputConstant;
+        }
         const requestLocation = requestLocationMap[location];
-        const kind: InputOperationParameterKind =
-            InputOperationParameterKind.Method;
+        const isContentType: boolean =
+            requestLocation === RequestLocation.Header &&
+            name.toLowerCase() === "content-type";
+        const kind: InputOperationParameterKind = isContentType
+            ? InputOperationParameterKind.Constant
+            : InputOperationParameterKind.Method;
         return {
-            Name: name,
+            Name: param.name,
```

Comment:
What's the difference between `name` and `param.name`?

Reply:
`param.name` will remain the same as the parameter defined. but cadl compile will change the `name` according to name convention. e.g. `@header contentType`, cadl compile will change the `name` to 'content-type', and remain the `param.name` as defined